nixos/tests/kanata.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, ... }:
+
+{
+  name = "kanata";
+
+  meta = {
+    maintainers = with lib.maintainers; [ linj ];
+  };
+
+  nodes = {
+    machine = {
+      specialisation.withKanata.configuration = {
+        services.kanata = {
+          enable = true;
+          keyboards.all = {
+            config = ''
+              ;; map 1 to 2
+              (defsrc 1)
+              (deflayer default-layer 2)
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  testScript = ''
+    machine.wait_for_unit("default.target")
+
+    with subtest("enabling kanata at runtime, i.e., nixos-rebuild switch"):
+      machine.succeed("/run/current-system/specialisation/withKanata/bin/switch-to-configuration switch")
```

Comment:
Why do we need a specialization here?

Reply:
Because the point is that this bug only happens this way.  If you directly boot into this profile, it works fine.

Note that this is a minimal reproducer for https://github.com/NixOS/nixpkgs/issues/317282 which is caused by https://github.com/NixOS/nixpkgs/issues/334017.

Reply:
Then this should be definitely a comment otherwise the next person is going to ask the themselves the same question.

Reply:
Also I think we should wait for a respone from someone that maintains systemd to not collect a known failing test which then just sits around and is broken.

Reply:
> Then this should be definitely a comment otherwise the next person is going to ask the themselves the same question.

Good idea.  I have added a comment.


> I think we should wait for a respone from someone that maintains systemd to not collect a known failing test which then just sits around and is broken.

I have linked this PR in https://github.com/NixOS/nixpkgs/issues/334017 and pinged @Nixos/Systemd and @nikstur (author of `sysinit-reactivation.target`) there.